### PR TITLE
Added a test case to compare group layer filtering

### PR DESCRIPTION
**Pull request by mastef**
Related to #52

### CODE REVIEW

File: test/SvgGenerationTest.hx [hunk 1]
```diff
@@ -134,9 +140,16 @@ class SvgGenerationTest
         var pngName = generatePng(svgName, pngWidth, pngHeight);
         compareGeneratedToExpected(svgName, pngName);
     }
-    
+
+    private function generateAndCompareTo(svgName1:String, svgName2:String)
+    {
+        var pngName1 = generatePng(svgName1);
+        var pngName2 = generatePng(svgName2, 0, 0, "red");
```
Comment (ashes999) [inline test/SvgGenerationTest.hx]: Shouldn't `"red"` also be a function parameter?

File: test/SvgGenerationTest.hx [hunk 2]
```diff
@@ -175,7 +189,7 @@ class SvgGenerationTest
         var backgroundColor = 0x00FFFFFF;
         var shape = new Shape();
         // scale/render the SVG to this size
-        svg.render(shape.graphics, 0, 0, width, height);
+        svg.render(shape.graphics, 0, 0, width, height, layerid);
```
Comment (ashes999) [inline test/SvgGenerationTest.hx]: Do you need a null-check here if `layerid` isn't null?

File: test/SvgGenerationTest.hx [hunk 3]
```diff
@@ -146,6 +159,7 @@ class SvgGenerationTest
         // (this makes it easier to see in the report).
         var width:Int = pngWidth;
         var height:Int = pngHeight;
+        var layerid:String = pLayerID;
```
Comment (ashes999) [inline test/SvgGenerationTest.hx]: Should this be `layerId`?

File: test/SvgGenerationTest.hx [hunk 4]
```diff
@@ -269,6 +283,75 @@ class SvgGenerationTest
         }
 	}
     
+    // Compares 2 generated PNGs
+    private function compareGeneratedPNGs(pngName1:String, pngName2:String)
```
Comment (ashes999) [inline test/SvgGenerationTest.hx]: I'm not really crazy about the amount of copy/paste here. If we decide to change our pixel-diff-calculation later, or find a bug in this code, we have to remember to fix it in two places.

Could you please refactor out the common code instead? Or, if that makes the code too complex or hard to understand, what else can we do here?

**Comment by ashes999** [comment]
I added some comments. Mostly looks good, I just have some concerns about copy/pasting a large and complicated bunch of code.

@jgranick will probably weigh in here as the official reviewer though.

File: test/SvgGenerationTest.hx (hunk 2 above)
Comment (mastef) [inline test/SvgGenerationTest.hx]: `layerid == null` is handled in the lib

File: test/SvgGenerationTest.hx (hunk 4 above)
Comment (mastef) [inline test/SvgGenerationTest.hx]: Unified the logic and moved to `comparePixels()`

File: test/SvgGenerationTest.hx (hunk 1 above)
Comment (mastef) [inline test/SvgGenerationTest.hx]: Changed to `generateAndCompareWithLayerFilter(svgName1:String, layerId1:String, svgName2:String, layerId2:String)` with layerId as parameters

File: test/SvgGenerationTest.hx (hunk 3 above)
Comment (mastef) [inline test/SvgGenerationTest.hx]: Changed to `layerId`

**Comment by ashes999** [comment]
Final review/merge @jgranick please?

**Pull request merged**